backend/src/nodes/impl/pytorch/auto_split.py

```diff
@@ -72,9 +72,13 @@ def upscale(img: np.ndarray, _: object):
         try:
             # convert to tensor
             img = np.ascontiguousarray(img)
-            if not img.flags.writeable:
+            if not img.flags.writeable and device == torch.device("cpu"):
                 img = np.copy(img)
+            else:
+                # since we are going to copy the image to the GPU, we can skip the copy here
+                img.flags.writeable = True
             input_tensor = torch.from_numpy(img).to(device, dtype)
+            img.flags.writeable = False
```

Comment:
This is quite complex, so please move this into a function.

It's also not 100% correct. If `from_numpy` or `to` throws, the image might be incorrectly flagged as writeable.